virttest/qemu_vm.py

```diff
@@ -2390,41 +2259,19 @@ def _create_serial_console(self):
         Let's consider the first serial port as serial console.
         Note: requires a version of netcat that supports -U
         """
-        try:
-            tmp_serial = self.serial_ports[0]
-        except IndexError:
+        if self.serial_session_device is None:
             logging.warning("No serial ports defined!")
             return
-        log_name = "serial-%s-%s.log" % (tmp_serial, self.name)
+        log_name = "serial-%s-%s.log" % (
+            self.serial_session_device, self.name)
         self.serial_console_log = os.path.join(utils_misc.get_log_file_dir(),
                                                log_name)
         self.serial_console = aexpect.ShellSession(
-            "nc -U %s" % self.get_serial_console_filename(tmp_serial),
+            "nc -U %s" % self.serial_console_log,
```

Comment:
`serial_console_log` is the log path + log file name, it is used to save the serial log.
`get_serial_console_filename(serial_session_device)` is the `path` in qemu command line.
What we can use for 'nc' is the `path` in qemu command line,  not log file name.